src/main/kotlin/pl/edu/uj/ii/ksi/mordor/services/crawlers/MetadataCrawler.kt

```diff
@@ -1,34 +1,45 @@
-package pl.edu.uj.ii.ksi.mordor.services
+package pl.edu.uj.ii.ksi.mordor.services.crawlers
 
 import java.io.File
 import org.slf4j.LoggerFactory
 import org.springframework.beans.factory.annotation.Value
 import org.springframework.scheduling.annotation.Scheduled
 import org.springframework.stereotype.Service
 import pl.edu.uj.ii.ksi.mordor.persistence.repositories.FileEntryRepository
+import pl.edu.uj.ii.ksi.mordor.services.ExternalUserService
+import pl.edu.uj.ii.ksi.mordor.services.FileEntryCreator
 import pl.edu.uj.ii.ksi.mordor.services.repository.RepositoryService
 
 @Service
 class MetadataCrawler(
     @Value("\${mordor.root_path}") private val rootPathStr: String,
-    val repositoryService: RepositoryService,
-    val fileEntryRepository: FileEntryRepository,
-    val fileEntryCreator: FileEntryCreator
-) {
+    private val repositoryService: RepositoryService,
+    private val fileEntryRepository: FileEntryRepository,
+    private val fileEntryCreator: FileEntryCreator
 
+) {
     companion object {
         private val logger = LoggerFactory.getLogger(ExternalUserService::class.java)
+        var progress = CrawlerProgress()
     }
 
     @Scheduled(fixedDelay = 60 * 1000 * 1000)
+    @Suppress("MagicNumber")
     fun crawl() {
```

Comment:
Since you are cleaning up the crawler class you could as well add some basic multi thread handling (allow only one instance of the crawler to run at once)